### DISCUSSION TITLE
Known ongoing incident with stcelo.xyz web app (Resolved)

### DISCUSSION THREAD

**Discussion by arthurgousset** [celo-org/staked-celo-web-app#138] [Q&A] [answered]
Hi all 👋

## TLDR

- There is a known ongoing incident that affects [stcelo.xyz](https://stcelo.xyz/) users that connect using WalletConnect.
- This is related to the [shutdown of WalletConnect v1 and migration to WalletConnect v2 ](https://forum.celo.org/t/walletconnect-v1-to-v2-migration/5184)today.

## Live updates

**Status: Resolved**  ✅

We will keep this Github Discussion updated, so wallet developers and the Celo community can follow along.

We apologise for the inconvenience and are working hard to bring back WalletConnect support for the [stcelo.xyz](https://stcelo.xyz/) app. Thank you for your patience! 🙏

**Comment by dckesler** [comment]
This appears to be an issue other people are struggling with in WalletConnect v2 right now. We're actively following [this discussion](https://github.com/orgs/WalletConnect/discussions/2731) now to see what the WalletConnect team suggests.

**Comment by nicolasbrugneaux** [comment]
## Update Monday, July 3rd

Hello there, we are still hard at working trying to find a solution to this issue. The WalletConnect connecting issues seems widespread enough to impact users from a lot of chains and wallets. It seems however they're hard at work trying to fix these and we're in contact with them on their [official discord](https://discord.com/channels/492410046307631105/1105316298390511626) and through the github discussion [linked above](https://github.com/orgs/WalletConnect/discussions/2731).

While we're aware it doesn't help you at the time being, please know we're doing the best we can to get the ball rolling and get a fix ASAP. Furthermore, thanks to @jeanregisser we [improved our WCv2 support](https://github.com/celo-org/rainbowkit-celo) with rainbowkit, and the StakedCelo v2 that should be released July 11th  just happens to use rainbowkit. We will keep you inform of further progress.

Edit: additional reading on [debugging with the WC team here](https://discord.com/channels/492410046307631105/1105316298390511626/threads/1125448565569900646)

**Comment by nicolasbrugneaux** [comment]
## Update Tuesday, July 4th (morning CET)

Hello there,

today's was marked by some good development, but still no definitive fix. 


With the help of the WalletConnect team, we are debugging what seems to be a widespread issues with all wallets using `wagmi` under the hood, a [PR was already merged](https://github.com/wagmi-dev/references/pull/396) and [published](https://github.com/wagmi-dev/references/releases/tag/%40wagmi%2Fconnectors%402.6.4) but the source of our problems: the [`Failed to decrypt` error](https://github.com/orgs/WalletConnect/discussions/2731) still happens in production environments. We also have a [direct thread with the WC team](https://discord.com/channels/492410046307631105/1125448565569900646) and we are optimistic as to a fast resolution.


Wagmi PR:
- https://github.com/wagmi-dev/references/pull/396

Walletconnect discussion:
> - [Uncaught (in promise) Error: Failed to decrypt](https://github.com/orgs/WalletConnect/discussions/2731) _WalletConnect#2731_

Discord thread:
> - [Thread between Nicolas Brugneaux and glitch, a WalletConnect maintainer](https://discord.com/channels/492410046307631105/1125448565569900646) Discord

**Reply by nicolasbrugneaux** [reply]
Some more insight regarding the `wagmi` issue: it seems one of the causes for issues is the fact that `wagmi` implicitely uses the namespaces passed to the provider in order. Here is some pseudocode to illustrate the issue:

We were using `chains: [celoAlfajores, celoCannoli, celo]` 
which _roughly_ translate `{requiredChain: celoAlfajores, optionalChains: [celoCannoli, celo]` from wagmi's point of view

As you can see this is flawed and `celo` (our mainnet) should be first in that array.

While this _still_ doesn't fix the `Failed to decrypt`, it at least fixes another issue that would have been caused by wallets only supporting mainnet and not the testnets.

**Comment by nicolasbrugneaux** [comment]
## Update Tuesday, July 4th (evening CET)

Hello again,

the issue is now **mitigated** through a temporary fix in the following PR:
- https://github.com/celo-org/staked-celo-web-app/pull/140

It means the incident is resolved but we had to compromise on performances. It seems either rainbowkit, wagmi, or walletconnect have an issue when minified and optimized through webpack. We disabled these optimizations in order to re-establish WalletConnect capacities. We'll keep investigating to re-enable thee optimizations in order to have a better UX.

Thank you all for your understanding

**Comment by nicolasbrugneaux** [comment, marked as answer]
## Update Thursday, July 6th

Hello again,

the issue is now resolved with minification re-enabled through the following PR:
- https://github.com/celo-org/staked-celo-web-app/pull/141

It was due to a bug in the [SWC](https://swc.rs/) minifier. Using Babel's minifier instead solves the issue completely.
